src/Widgets/DiskBar.vala

```diff
@@ -24,91 +24,69 @@ public class Installer.DiskBar: Gtk.Grid {
     public uint64 size { get; construct; }
     public Gee.ArrayList<PartitionBar> partitions { get; construct; }
 
-    public Gtk.Grid label { get; private set; }
-    private Gtk.Grid legend_container;
+    public Gtk.Box label;
+    private Gtk.Box bar;
+    private Gtk.ScrolledWindow legend;
+    private Gtk.Box legend_container;
+    private uint64 unused;
 
-    public DiskBar (string disk_name, string disk_path, uint64 size, Gee.ArrayList<PartitionBar> partitions) {
+    public const int MIN_PARTITION_WIDTH = 28;
+
+    public DiskBar (
+        string model,
+        string path,
+        uint64 size,
+        Gee.ArrayList<PartitionBar> partitions
+    ) {
         Object (
-            disk_name: disk_name,
-            disk_path: disk_path,
+            disk_name: model,
+            disk_path: path,
             partitions: partitions,
             size: size
         );
     }
 
     construct {
-        var name_label = new Gtk.Label ("<b>%s</b>".printf (disk_name));
-        name_label.halign = Gtk.Align.END;
-        name_label.use_markup = true;
-
-        var size_label = new Gtk.Label ("<small>%s %s</small>".printf (disk_path, GLib.format_size (size)));
-        size_label.halign = Gtk.Align.END;
-        size_label.use_markup = true;
-        size_label.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
-
-        label = new Gtk.Grid ();
-        label.orientation = Gtk.Orientation.VERTICAL;
-        label.row_spacing = 6;
-        label.margin_end = 12;
-        label.valign = Gtk.Align.CENTER;
-        label.add (name_label);
-        label.add (size_label);
+        unused = get_unused ();
+        generate_label ();
+        generate_bar ();
+        generate_legend ();
+
+        this.hexpand = true;
```

Comment:
Probably don't need to explicitly namespace these with `this`, it's clear without.